open-craft/custom-student-verification
# feat: send email to the user on ID verification

Uses edx-ace to send emails on student verification application
rejection and approval.

Related Tickets: [BB-4221]()

#### Testing Instructions

1. On a `juniper` devstack, clone this under `{devstack_folder}/src` (devstack_folder is the one that contains edx-platform, devstack, and the other folders.
1. Get into the LMS shell using `make lms-shell`.
1. Install the app using pip: `pip install -e /edx/src/custom-student-verification`
2. Restart LMS
3. Log into LMS with an unverified user and visit http://localhost:18000/custom-student-verification
4. Upload an image and submit
5. With a staff user logged in visit http://localhost:18000/admin/custom_student_verification/studentverificationrequest/
6. Open the verification request created above, mark it `approved` and save.
7. Verify that an email is sent by checking the latest file in `/edx/src/ace_messages` in `lms-shell`
8. Open the verification request created above again in your browser.
9. Mark the request `rejected` and save.
10. Verify that a rejected email is send by checking the file in `/edx/src/ace_messages`.
11. The above emails can also be seen in `lms-logs`

#### Author Notes
1. The email will be sent every time the verification request is marked `rejected`. This includes just opening a `rejected` request and saving it again from browser.
2. When the request is marked `rejected`, the `approved` records from `ManualVerification` are deleted.
3. The requirements are updated since `edx_ace` is added.
